doc/source/whatsnew/v1.3.0.rst

```diff
@@ -647,6 +647,7 @@ Deprecations
 - Deprecated setting :attr:`Categorical._codes`, create a new :class:`Categorical` with the desired codes instead (:issue:`40606`)
 - Deprecated behavior of :meth:`DatetimeIndex.union` with mixed timezones; in a future version both will be cast to UTC instead of object dtype (:issue:`39328`)
 - Deprecated using ``usecols`` with out of bounds indices for ``read_csv`` with ``engine="c"`` (:issue:`25623`)
+- Deprecated passing arguments as positional (except for ``by``) in :meth:`DataFrame.sort_values` and :meth:`Series.sort_values` (:issue:`41485`)
```

Comment:
this change won't solve the `error: Signature of "sort_values" incompatible with supertype "NDFrame"  [override]`. maybe we could also give `by` a default value (the dataframe columns). The 'by' keyword could still be passed positionally if it has a default value.

Reply:
we could also perhaps deprecate the axis keyword on Series at the same time to clean this up further.

Reply:
> we could also perhaps deprecate the axis keyword on Series at the same time to clean this up further.

I don't understand, wouldn't this be incompatible with the supertype?

> maybe we could also give by a default value (the dataframe columns). The 'by' keyword could still be passed positionally if it has a default value.

Sure, could that - that could be done directly in 1.3 without a prior warning, right? As current behaviour would be preserved

Reply:
> maybe we could also give by a default value (the dataframe columns). The 'by' keyword could still be passed positionally if it has a default value.

That might be a good idea, but I would personally leave that for a separate PR (as it seems an independent new feature)